Riot/Modules/Room/NotificationSettings/SwiftUI/FormPickerItem.swift

```diff
@@ -25,23 +25,21 @@ struct FormPickerItem: View {
     
     var title: String
     var selected: Bool
-    var onClick: ClickCallback?
+    var onClick: ClickCallback
     
     var body: some View {
-        HStack {
-            Text(title)
-                .font(Font(theme.fonts.body))
-                .foregroundColor(Color(theme.textPrimaryColor))
-            Spacer()
-            if selected {
+        Button(action: onClick) {
+            HStack {
+                Text(title)
+                    .font(Font(theme.fonts.body))
+                    .foregroundColor(Color(theme.textPrimaryColor))
+                Spacer()
                 Image("checkmark")
                     .foregroundColor(Color(theme.tintColor))
+                    .opacity(selected ? 1 : 0)
```

Comment:
Didn't particularly mean to commit that, but it's more readable I guess 🙃

Reply:
Opacity will visually hide it but still "take up the space" so if the text on the left grew you would see a blank space there. But agree with you the `if` syntax doesn't really fit with the declarative style. I think we could create a visibility modifier that allows us to control wether we want a view, `visible`, `hidden` (but "take up space") and `gone`(not added).